xbmc/utils/MathUtils.h

```diff
@@ -194,10 +194,10 @@ namespace MathUtils
    * the standard operator== since very close numbers might have different
    * representations.
    */
-  template<typename FloatT>
-  inline bool FloatEquals(FloatT f1, FloatT f2, FloatT maxDelta)
+  template<typename FloatL, typename FloatR, typename FloatD>
+  inline bool FloatEquals(FloatL f1, FloatR f2, FloatD maxDelta)
   {
-    return (std::abs(f2 - f1) < maxDelta);
+    return (std::abs(f2 - f1) <= maxDelta);
```

Comment:
`std::abs` returns double, probably you want `std::fabs`

`<` is better than `<=` because for same reason not makes much sense compare result of fabs with == 0.01f

Reply:
Some arguments for this function come as `double` triggering `-Wdouble-promotion` error minimum in EDL test. I guess I'll rename the function and typenames to reflect that fact.

Reply:
Unfortunately, `fabs` takes only floats while in our case every argument of the new function can be of any numeric type (for example, `long long int`, `float` and `double`). I guess I have to rename the function to remove the ambiguity.

Reply:
Then maybe leave as is:

```
  template<typename FloatT>
  inline bool FloatEquals(FloatT f1, FloatT f2, FloatT maxDelta)
```

template<typename FloatT>  guarantees that the 3 parameters are of the same type (all float or all double).

for `long long int`, etc. not makes sense use a comparison function but standard `==` anyway

Reply:
The issue is as follows: all EDL functions make float rounds then multiply by 1000 so a rounding error becomes exactly 1 (5299 vs 5300). So I decided to decouple the param value types to avoid `(float)GetEdlFor…` casts and introduced `<=` to handle that specific case. An alternative could be `1.001f` to ensure floats but as you see, there are arithmetic expressions calculated as `double` so single `FloatT` just does not work for every possible argument combination :(

Reply:
if `FloatEquals` it does not serve the purpose that is required here use something else....  but not hacking FloatEquals

Reply:
something else like what? Tgis is what I dont understand atm :(